### PR TITLE
[19] [UI] As a logged in user, I can see the complete survey screen

### LINKED ISSUE #19: [UI] As a logged in user, I can see the complete survey screen
## Why

- Show complete survey screen when the user complete survey
 
## Acceptance Criteria

- Show a confetti animation when the user complete the survey.
- Show a confetti animation for about 3 seconds and then dismiss.
Screen Shot 2021-06-04 at 9 41 17 AM

### PR DESCRIPTION
**Pull request by chiiida**
Resolves https://github.com/llleyelll/ic-surveys-ios/issues/19

## What happened 👀

- Implement a complete survey Lottie animation.
 
## Insight 📝

- Show Lottie animation when submitting survey and dismiss after a duration.

### CODE REVIEW

File: SurveyApp/Views/Lottie/CompleteSurveyView.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }
+    
+    private func setUpView() {
+        backgroundColor = .black
+        
+        textLabel.text = Localize.lottieCompleteSurvey()
+        textLabel.textColor = .white
+        textLabel.textAlignment = .center
+        textLabel.font = UIFont.bold(ofSize: .largerTitle)
```
Comment (blyscuit) [inline SurveyApp/Views/Lottie/CompleteSurveyView.swift]:
```suggestion
        textLabel.font = .bold(ofSize: .largerTitle)
```

File: SurveyApp/Extensions/UIKit/UIView/UIView+Present.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import UIKit
+
+extension UIView {
+    
+    func presentViewOnTopWindow(_ duration: TimeInterval = 0.3, for dismissAfter: TimeInterval = 0, _ completion: EmptyCompletion? = nil) {
+        alpha = 0
+        frame = UIScreen.main.bounds
+        let windows = UIApplication.shared.windows.first { $0.isKeyWindow }
+        windows?.addSubview(self)
```
Comment (markgravity) [inline SurveyApp/Extensions/UIKit/UIView/UIView+Present.swift]: Stop everything if `window` is not available
```suggestion
        guard let window = UIApplication.shared.windows.first ({ $0.isKeyWindow }) else { return }
        alpha = 0
        frame = UIScreen.main.bounds
        window.addSubview(self)
```

File: SurveyApp/Extensions/UIKit/UIView/UIView+Present.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        UIView.animate(
+            withDuration: duration,
+            delay: 0,
+            options: .curveEaseInOut,
+            animations: { [weak self] in
+                self?.alpha = 1
+            },
+            completion: { _ in
+                if dismissAfter <= 0 {
```
Comment (markgravity) [inline SurveyApp/Extensions/UIKit/UIView/UIView+Present.swift]: I suggest using `guard` here. What do you think? 🤔

File: SurveyApp/Extensions/UIKit/UIViewController/UIViewController+Lottie.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by Chananchida F. on 7/11/21.
+//
+
+import UIKit
+
+protocol LottieShowable {
+    
+    func showCompleteSurveyView(completion: EmptyCompletion?)
+}
```
Comment (markgravity) [inline SurveyApp/Extensions/UIKit/UIViewController/UIViewController+Lottie.swift]: I don't think we need to create a common protocol for these Lottie animations, IMO 🤔 
Just simply use it only for the Complete Survey screen, what do you think?

File: SurveyApp/Modules/SurveyQuestion/SurveyQuestionPresenter.swift [hunk 5]
```diff
@@ -54,7 +54,9 @@ extension SurveyQuestionPresenter: SurveyQuestionInteractorOutput {
         if let questions = questions, questionCount < questions.count {
```
Comment (markgravity) [inline SurveyApp/Modules/SurveyQuestion/SurveyQuestionPresenter.swift]: I prefer to use `guard` here too 👍

Comment (chiiida) [reply SurveyApp/Modules/SurveyQuestion/SurveyQuestionPresenter.swift]: Updated in #81 🙇‍♀️

File: SurveyApp/Extensions/UIKit/UIViewController/UIViewController+Lottie.swift (hunk 4 above)
Comment (chiiida) [reply SurveyApp/Extensions/UIKit/UIViewController/UIViewController+Lottie.swift]: @markgravity Do you mean we can simply move show logic to `CompleteSurveyView` like this:
```Swift
extension CompleteSurveyView {
    
    func show(completion: EmptyCompletion? = nil) {
        presentViewOnTopWindow(for: 0.8) {
            completion?()
        }
    }
}
```

and in `SurveyQuestionPresenter` we can call it like this:
```Swift
func didSubmitSurvey(questionCount: Int) {
    ...
    let completeSurveyView = CompleteSurveyView()
    completeSurveyView.show {
        self.view?.popToRootViewController()
    }
}
```
please help suggest 🙏

**Review by blyscuit — APPROVED** [review]

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**